x/mongo/driver/description/topology_version.go

```diff
@@ -0,0 +1,29 @@
+// Copyright (C) MongoDB, Inc. 2017-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package description
+
+import (
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+// TopologyVersion represents a software version.
+type TopologyVersion struct {
+	ProcessID primitive.ObjectID
+	Counter   int64
+}
+
+// MoreRecentThan returns if this TopologyVersion is more recent than the one passed in
+func (tv *TopologyVersion) MoreRecentThan(other *TopologyVersion) bool {
```

Comment:
Would it make more sense to have a general `CompareTopologyVersions(tv1, tv2 *TopologyVersion)` function? This would make the style consistent with `primitive.CompareTimestamps` and `session.MaxClusterTime`.

Reply:
Spoke about this offline. A general `CompareToplogyVersions` function doesn't make sense because there's no total ordering so you could have cases where flipping the arguments returns the same result. This still needs to be renamed though, because it's assuming that we'll call it as `currentVersion.MoreRecentThan(incomingResponse)`. If we called it the other way, it would give the wrong answer if the `ProcessID` fields aren't the same because it would say that the incoming response is older, when the spec says that it should be considered newer.